### PR TITLE
Header downloader - step 1

### PR DESCRIPTION
**Pull request by mriccobene**
This partially implement #297

Here we have the "responder" part of the downloader: it is able to connect to the sentry and respond to external request.
Also we have almost all packets (with test) and rpcs.

### CODE REVIEW

**Comment by Giulio2002** [comment]
I think you should change the stage1 directory name to downloader, i think we might want to reuse it to put all of the future work to be done on block bodies.

File: cmake/Hunter/core_packages.cmake [hunk 1]
```diff
@@ -19,3 +19,5 @@ hunter_add_package(intx)
 hunter_add_package(Microsoft.GSL)
 hunter_add_package(nlohmann_json)
 hunter_add_package(ethash)
+hunter_add_package(gRPC)
+hunter_add_package(Protobuf)
```
Comment (yperbasis) [inline cmake/Hunter/core_packages.cmake:23]: These 2 should go into `extra_packages`, not `core_packages`. Silkworm core is compatible with WebAssembly, while gRPC and Protobuf are probably not.

Comment (AndreaLanfranchi) [reply cmake/Hunter/core_packages.cmake:23]: Yeah right.

**Review by yperbasis — CHANGES_REQUESTED** [review]
I agree with @Giulio2002 : we should put stage 1 code alongside the rest of silkworm db. "silkworm db" is a bit unfortunate (and we should probably rename it) but basically "silkworm db" means everything else but "silkworm core" (Silkworm core primarily deals with the EVM execution logic and is compatible with WebAssembly, while "silkworm db" is not  compatible with WebAssembly).

For example, instead of adding
```
stages/stage1/messages/InboundGetBlockBodies.hpp
```
add something like
```
db/downloader/messages/inbound_get_block_bodies.hpp
```
Please spend some time ensuring that your code fits nicely into existing "silkworm db" code.  Please follow the conventions in https://github.com/torquem-ch/silkworm#code-style

As to protobuf-generated files, it's probably OK for them to sit in the `interfaces` directory, no need to move to db.

**Comment by mriccobene** [comment]
Done:
- [x] moved code to silkworm/db
- [x] added build target for test in silkworm/db - all test (47) pass, 10 are about the downloader
- [x] fixed some code-style issues
- [x] removed c++20 dependencies - refactored the entire rlp::encode overload set (of the downloader) - tested with gcc 10.2

Todo:
- [ ] fix circle/ci issues
- [ ] fix MSVC build issues
- [ ] check clang compile/build

Future improvements of the code base:
- rework the rlp::encode function template overload set (remove template?) and avoid code duplication [encode(vector), encode(packet66)]
- add interfaces as submodule https://github.com/ledgerwatch/interfaces

Please consider this post that can help the reviewing process https://github.com/torquem-ch/silkworm/issues/297#issuecomment-889698577

File: appveyor.yml [hunk 2]
```diff
@@ -32,7 +32,7 @@ before_build:
   - git submodule sync
   - git submodule update --init --recursive
   - SET INCLUDE=C:\Tools\vcpkg\installed\x64-windows\include;%INCLUDE%
-  - SET PATH=C:\Tools\vcpkg\installed\x64-windows\bin;%PATH%
+  - SET PATH=C:\Tools\vcpkg\installed\x64-windows\bin;C:\Strawberry\perl\bin;%PATH%
```
Comment (yperbasis) [inline appveyor.yml:35]: Why is perl required?

File: cmd/CMakeLists.txt [hunk 3]
```diff
@@ -106,6 +106,12 @@ if(NOT SILKWORM_CORE_ONLY)
   add_executable(trie trie.cpp)
   target_link_libraries(trie PRIVATE silkworm_db CLI11::CLI11)
 
+  find_package(gRPC CONFIG REQUIRED)
```
Comment (yperbasis) [inline cmd/CMakeLists.txt:109]: No need to find gRPC since it's a public dependency of silkworm_db.

File: cmd/CMakeLists.txt [hunk 4]
```diff
@@ -106,6 +106,12 @@ if(NOT SILKWORM_CORE_ONLY)
...
+  add_executable(download_headers download_headers.cpp)
+  target_link_libraries(download_headers PRIVATE silkworm_db silkworm_core CLI11::CLI11)
+  target_include_directories(download_headers PRIVATE ${CMAKE_SOURCE_DIR})
+  set_property(TARGET download_headers PROPERTY CXX_STANDARD 20)
```
Comment (yperbasis) [inline cmd/CMakeLists.txt:113]: Can we do w/o C++ 20 for download_headers?

File: appveyor.yml (hunk 2 above)
Comment (mriccobene) [reply appveyor.yml:35]: It is required to build OpenSSL that is a dependency of gRPC

**Review by yperbasis — APPROVED** [review]

**Pull request merged**